### PR TITLE
specify endpoint url for SQS to support PrivateLink

### PR DESCRIPTION
**Pull request by DaisukeMiyamoto**
*Description of changes:*
- specify endpoint on send-message for SQS in compute_ready.erb
  - In default, SQS command in AWS CLI use legacy endpoint url (*region*.queue.amazonaws.com). It prevent using PrivateLink url (sqs.*region*.amazonaws.com). Therefore this fix specifies new endpoint url.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by DaisukeMiyamoto** [comment]
This PR is related with other PR in aws-parallelcluster-node
https://github.com/aws/aws-parallelcluster-node/pull/208

File: templates/default/compute_ready.erb [hunk 1]
```diff
@@ -10,4 +10,4 @@ instance_type_url="http://169.254.169.254/latest/meta-data/instance-type"
...
 local_hostname_url="http://169.254.169.254/latest/meta-data/local-hostname"
 local_hostname=$(curl --retry 3 --retry-delay 0 --silent --fail ${local_hostname_url})
-<%= node['cfncluster']['cookbook_virtualenv_path'] %>/bin/aws --region ${cfn_region} sqs send-message --queue-url ${cfn_sqs_queue} --message-body '{"Type" : "Notification", "Message" : "{\"StatusCode\":\"Complete\",\"Description\":\"Succesfully launched '${instance_id}'\",\"Event\":\"parallelcluster:COMPUTE_READY\",\"EC2InstanceId\":\"'${instance_id}'\",\"Slots\":\"'${cfn_instance_slots}'\",\"LocalHostname\":\"'${local_hostname}'\",\"EC2InstanceType\":\"'${instance_type}'\"}"}'
+<%= node['cfncluster']['cookbook_virtualenv_path'] %>/bin/aws --region ${cfn_region} sqs send-message --queue-url ${cfn_sqs_queue} --endpoint https://sqs.${cfn_region}.amazonaws.com --message-body '{"Type" : "Notification", "Message" : "{\"StatusCode\":\"Complete\",\"Description\":\"Succesfully launched '${instance_id}'\",\"Event\":\"parallelcluster:COMPUTE_READY\",\"EC2InstanceId\":\"'${instance_id}'\",\"Slots\":\"'${cfn_instance_slots}'\",\"LocalHostname\":\"'${local_hostname}'\",\"EC2InstanceType\":\"'${instance_type}'\"}"}'
```
Comment (demartinofra) [inline templates/default/compute_ready.erb:13]: Thanks for the contribution. The PR looks good to me but we need to handle the SQS endpoint in China regions as well. In that case the domain is `amazonaws.com.cn`. 

We could either extract the full endpoint from ${cfn_sqs_queue} that looks something like `https://sqs.us-east-2.amazonaws.com/123456789012/ab1-MyQueue-A2BCDEF3GHI4` or add some conditional logic such as `$([[ ${_region} =~ ^cn- ]] && echo ".cn")`

File: templates/default/compute_ready.erb [hunk 2]
```diff
@@ -10,4 +10,4 @@ instance_type_url="http://169.254.169.254/latest/meta-data/instance-type"
...
 local_hostname=$(curl --retry 3 --retry-delay 0 --silent --fail ${local_hostname_url})
-<%= node['cfncluster']['cookbook_virtualenv_path'] %>/bin/aws --region ${cfn_region} sqs send-message --queue-url ${cfn_sqs_queue} --message-body '{"Type" : "Notification", "Message" : "{\"StatusCode\":\"Complete\",\"Description\":\"Succesfully launched '${instance_id}'\",\"Event\":\"parallelcluster:COMPUTE_READY\",\"EC2InstanceId\":\"'${instance_id}'\",\"Slots\":\"'${cfn_instance_slots}'\",\"LocalHostname\":\"'${local_hostname}'\",\"EC2InstanceType\":\"'${instance_type}'\"}"}'
+<%= node['cfncluster']['cookbook_virtualenv_path'] %>/bin/aws --region ${cfn_region} sqs send-message --queue-url ${cfn_sqs_queue} --endpoint https://sqs.${cfn_region}.amazonaws.com$([[ ${cfn_region} =~ ^cn- ]] && echo ".cn") --message-body '{"Type" : "Notification", "Message" : "{\"StatusCode\":\"Complete\",\"Description\":\"Succesfully launched '${instance_id}'\",\"Event\":\"parallelcluster:COMPUTE_READY\",\"EC2InstanceId\":\"'${instance_id}'\",\"Slots\":\"'${cfn_instance_slots}'\",\"LocalHostname\":\"'${local_hostname}'\",\"EC2InstanceType\":\"'${instance_type}'\"}"}'
```
Comment (demartinofra) [inline templates/default/compute_ready.erb:13]: are we sure this is not `--endpoint-url`?

Comment (DaisukeMiyamoto) [reply templates/default/compute_ready.erb]: Thanks for your checking.
Yes, you are definitely right. `--endpoint-url` is correct.
In my excuse, I found (and surprised) both `--endpoint` and `--endpoint-url` could work.
Anyway, I fixed it.

**Pull request merged**